cli/cmd/install.go

```diff
@@ -331,3 +348,69 @@ func readIntoBytes(filename string) ([]byte, error) {
 
 	return buf.Bytes(), nil
 }
+
+func globalConfig(options *installOptions) *config.GlobalConfig {
+	var identityContext *config.IdentityContext
+	if options.enableTLS() {
+		identityContext = &config.IdentityContext{}
+	}
+
+	return &config.GlobalConfig{
+		LinkerdNamespace: controlPlaneNamespace,
+		CniEnabled:       options.noInitContainer,
+		IdentityContext:  identityContext,
+	}
+}
+
+func proxyConfig(options *installOptions) *config.ProxyConfig {
+	ignoreInboundPorts := []*config.Port{}
+	for _, port := range options.ignoreInboundPorts {
+		ignoreInboundPorts = append(ignoreInboundPorts, &config.Port{Port: uint32(port)})
+	}
+
+	ignoreOutboundPorts := []*config.Port{}
+	for _, port := range options.ignoreOutboundPorts {
+		ignoreOutboundPorts = append(ignoreOutboundPorts, &config.Port{Port: uint32(port)})
+	}
+
+	return &config.ProxyConfig{
+		ProxyImage: &config.Image{
+			ImageName:  k8s.ProxyImageName,
+			PullPolicy: options.imagePullPolicy,
+			Registry:   options.dockerRegistry,
+		},
+		ProxyInitImage: &config.Image{
+			ImageName:  k8s.ProxyInitImageName,
+			PullPolicy: options.imagePullPolicy,
+			Registry:   options.dockerRegistry,
```

Comment:
For parity with the existing implementation, we unfortunately can't just use the value of the `--docker-registry` flag here. If the value of `--proxy-image` includes a non-default registry, we use that instead. For instance, if I ran:

```
linkerd install --registry foo.com --proxy-image bar.com/custom-proxy
```

That should produce:

```
&config.Image{
	ImageName: "custom-proxy",
	PullPolicy: "IfNotPresent",
	Registry: "bar.com",
}
```

But as written it uses "foo.com" for the registry.

Reply:
(sorry, hit submit too soon)

Same comment goes for the proxy-init image. We should probably refactor `taggedProxyImage` and `taggedProxyInitImage` to make the registry more accessible.

Reply:
Per discussion with @klingerf, we will remove the registry field to avoid having to workaround edge cases like fully-qualified image name, versus user-provided Linkerd version etc.